crates/bevy_ecs/src/reflect/mod.rs

```diff
@@ -49,6 +49,17 @@ impl DerefMut for AppTypeRegistry {
     }
 }
 
+impl AppTypeRegistry {
+    /// Creates [`AppTypeRegistry`] and calls [`register_derived_types`](TypeRegistry::register_derived_types) on it.
+    ///
+    /// See [`register_derived_types`](TypeRegistry::register_derived_types) for more details.
```

Comment:
Nit: I think we can avoid the duplicate reference to `register_derived_types` while also giving a bit more detail here.
```suggestion
    /// Creates [`AppTypeRegistry`] and automatically registers all types deriving [`Reflect`].
    ///
    /// See [`TypeRegistry::register_derived_types`] for more details.
```